projects/components/src/filtering/filter/parser/types/abstract-filter-parser.ts

```diff
@@ -0,0 +1,94 @@
+import { FilterAttribute } from '../../filter-attribute';
+import { FilterAttributeType } from '../../filter-attribute-type';
+import { FilterOperator, fromUrlFilterOperator, toUrlFilterOperator, UrlFilterOperator } from '../../filter-operators';
+import { ParsedFilter, SplitFilter } from '../parsed-filter';
+
+export abstract class AbstractFilterParser<TValue> {
+  public abstract supportedAttributeTypes(): FilterAttributeType[];
+  public abstract supportedOperators(): FilterOperator[];
+
+  protected abstract parseValueString(
+    attribute: FilterAttribute,
+    splitFilter: SplitFilter<FilterOperator>
+  ): TValue | undefined;
+
+  protected parseKeyString(attribute: FilterAttribute, splitFilter: SplitFilter<FilterOperator>): string | undefined {
+    return attribute.displayName.toLowerCase() !== splitFilter.lhs.toLowerCase() ? undefined : attribute.name;
```

Comment:
we use display for key? could we match either?

Reply:
Correct, the User filters all work with Display Name. The URL filters work with name. We could probably match on either, but I haven't tested that thoroughly enough to know if it will work in all cases. Rather leave as is for now and expand the check later if we have a reason to.